src/streaming.rs

```diff
@@ -1,91 +1,152 @@
 //! ABNF Core Rules (RFC5234 B.1.)
 
+use std::ops::{RangeFrom, RangeTo};
+
 use nom::{
-    branch::alt,
-    bytes::streaming::tag,
-    character::{
-        is_alphabetic, is_digit as nom_is_digit, is_hex_digit as nom_is_hex_digit,
-        streaming::line_ending,
-    },
-    IResult,
+    character::streaming::satisfy,
+    combinator::{opt, recognize},
+    error::ParseError,
+    multi::many0_count,
+    sequence::{pair, terminated},
+    AsChar, Err as OutCome, IResult, InputIter, InputLength, Needed, Offset, Slice,
 };
 
-/// A-Z / a-z
-///
-/// ALPHA = %x41-5A / %x61-7A
-pub fn is_alpha(byte: u8) -> bool {
-    is_alphabetic(byte)
+use crate::{
+    is_alpha, is_bit, is_char, is_cr, is_ctl, is_digit, is_dquote, is_hexdig, is_htab, is_lf,
+    is_sp, is_wsp,
+};
+
+/// ALPHA = %x41-5A / %x61-7A ; A-Z / a-z
+pub fn alpha<I, E>(input: I) -> IResult<I, char, E>
+where
+    I: InputLength + InputIter + Slice<RangeFrom<usize>> + Clone,
```

Comment:
This really bothers me. I know why this is required, but still it makes the function signature really, really complicated. Do we have any alternative/more readable solution?

Reply:
not really that the price to pay, we could use trait alias I think but I that about all.

Reply:
Okay, so let's go with that for now.

Reply:
worth saying that anyway 99% of other nom parser look like this so a user of nom shouldn't have much problem understand this.